cli/src/lib.rs

```diff
@@ -513,35 +513,44 @@ fn override_toolchain(cfg_override: &ConfigOverride) -> Result<RestoreToolchainC
             Ok(version)
         }
 
+        fn is_solana_version_installed(version: &str) -> Result<bool> {
+            let output = std::process::Command::new("solana").arg("--version").output()?;
+            let output_version = std::str::from_utf8(&output.stdout)?;
+            Ok(output_version.contains(version))
+        }
+
         if let Some(solana_version) = &cfg.toolchain.solana_version {
             let current_version = get_current_version("solana")?;
             if solana_version != &current_version {
                 // We are overriding with `solana-install` command instead of using the binaries
                 // from `~/.local/share/solana/install/releases` because we use multiple Solana
                 // binaries in various commands.
-                fn override_solana_version(version: String) -> std::io::Result<bool> {
-                    std::process::Command::new("solana-install")
-                        .arg("init")
-                        .arg(&version)
-                        .stderr(Stdio::null())
-                        .stdout(Stdio::null())
-                        .spawn()?
-                        .wait()
-                        .map(|status| status.success())
-                }
+                if !is_solana_version_installed(solana_version)? {
```

Comment:
This expression will always be `true` here since we're checking 

```rs
if solana_version != &current_version {
```

above.

These checks will always return the same value because `is_solana_version_installed` implementation only checks the current version, which makes this check redundant here.